logstash-core/lib/logstash/agent.rb

```diff
@@ -235,15 +236,15 @@ def collect_metrics?
   def create_pipeline(settings, config=nil)
     if config.nil?
       begin
-        config = fetch_config(settings)
+        config = fetch_config
       rescue => e
         @logger.error("failed to fetch pipeline configuration", :message => e.message)
         return
       end
     end
 
     begin
-      LogStash::Pipeline.new(config, settings, metric)
+      LogStash::Pipeline.new(config.config_string, config.settings, metric)
```

Comment:
The problem is, If I change the signature of the Pipeline this will make plugins test to fails.